neo4j-to-neptune/src/main/java/com/amazonaws/services/neptune/util/NeptuneBulkLoader.java

```diff
@@ -0,0 +1,397 @@
+/*
+Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+A copy of the License is located at
+    http://www.apache.org/licenses/LICENSE-2.0
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+*/
+
+package com.amazonaws.services.neptune.util;
+
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+import java.io.File;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * Utility class for uploading local CSV files to Amazon S3 and loading them into Neptune
+ * Designed for Neptune data loading workflows with bulk loading capability
+ */
+public class NeptuneBulkLoader implements AutoCloseable {
+
+    private static final Set<String> BULK_LOAD_STATUS_CODES_COMPLETED = Set.of(
```

Comment:
Note: `Set.of()` was introduced in Java 9 and this project still builds on min Java 8. There should potentially be a discussion around dropping Java 8 but as of right now this is unsupported.